tycho-its/src/test/java/org/eclipse/tycho/test/baseline/BaselineMojoTest.java

```diff
@@ -139,4 +196,24 @@ private Verifier getBaselineProject(String project) throws Exception {
 		verifier.addCliOption(project + "/pom.xml");
 		return verifier;
 	}
+
+	private static void bumpPluginProjectVersion(Path projectRoot, String from, String to) throws IOException {
+		replaceStringInFile(projectRoot.resolve("META-INF/MANIFEST.MF"), //
+				"Bundle-Version: " + from, "Bundle-Version: " + to);
+		UnaryOperator<String> mavenVersion = v -> """
+				\t<artifactId>my.api.bundle</artifactId>
+				\t<version>""" + v + "</version>";
+		replaceStringInFile(projectRoot.resolve("pom.xml"), mavenVersion.apply(from), mavenVersion.apply(to));
+	}
+
+	private static void replaceStringInFile(Path file, String target, String replacement) throws IOException {
+		if (!"\n".equals(System.lineSeparator())) { // Text-Blocks only use \n
```

Comment:
I don't think this is a good idea, the method should just do the replacement and the caller should take care of do the replacement.

Reply:
The problem is that TextBlocks always use `\n` to separate newlines. Therefore on Windows they cannot match the content of the checkout files, which are usually checkedout using `\r\n` and at the same time, the replacement should also use \r\n on Windows. This ensures that the content of the generated files is the same as if they were all committed to the git repo.

I split this method into two so that we can have a 'pure' version and one that does this nasty replacement, but avoids code repetition in the callers.

Reply:
Then textblocks are probably not the right choice here ;-)
It defeats a bit their purpose if we need to handle them special afterwards and one should then avoid them in this context.

Reply:
Done.